creator-node/src/snapbackSM.js

```diff
@@ -47,63 +61,382 @@ class SnapbackSM {
+      logger.error(`Invalid sync data found`)
+      logger.error(job.data)
+      return
+    }
+    const syncWallet = syncRequestParameters.data.wallet[0]
+    const primaryClockValue = await this.getUserPrimaryClockValue(syncWallet)
+    const secondaryUrl = syncRequestParameters.baseURL
+    this.log(`------------------Process SYNC | User ${syncWallet} | Target: ${secondaryUrl} ------------------`)
+
+    // Issue sync request to secondary
+    await axios(syncRequestParameters)
+
+    // Monitor the sync status
+    await this.monitorSecondarySync(syncWallet, primaryClockValue, secondaryUrl)
+
+    // Exit when sync status is computed
+    // Determine how many times to retry this operation
+    this.log('------------------END Process SYNC------------------')
+  }
+
+  // Function which ensures that state machine has been initialized correctly
+  // If not available on startup, subsequent state machine will attempt to initialize until success
+  async initializeNodeIdentityConfig () {
+    this.log(`Initializing SnapbackSM`)
+    this.log(`Retrieving spID for ${this.endpoint}`)
+    const recoveredSpID = await this.audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceProviderIdFromEndpoint(
+      this.endpoint
+    )
+    config.set('spID', recoveredSpID)
+    // A returned spID of 0 means this this.endpoint is currently not registered on chain
+    // In this case, the stateMachine is considered to be 'uninitialized'
+    if (recoveredSpID === 0) {
+      this.initialized = false
+      this.log(`Failed to recover spID for ${this.endpoint}, received ${config.get('spID')}`)
+      return
+    }
+    this.log(`Recovered ${config.get('spID')} for ${this.endpoint}`)
+    this.initialized = true
+    return this.initialized
+  }
+
+  // Initialize the state machine
+  async init () {
+    await this.stateMachineQueue.empty()
+    await this.syncQueue.empty()
+
+    const isUserMetadata = config.get('isUserMetadataNode')
+    if (isUserMetadata) {
+      this.log(`SnapbackSM disabled for userMetadataNode. ${this.endpoint}, isUserMetadata=${isUserMetadata}`)
+      return
     }
+
+    // Initialize state machine queue processor
+    this.stateMachineQueue.process(
+      async (job, done) => {
+        try {
+          await this.processStateMachineOperation(job)
+        } catch (e) {
+          this.log(`stateMachineQueue error processing ${e}`)
+        } finally {
+          // TODO: Remove dev mode
+          if (!config.get('triggerSyncOnWrite')) {
+            this.log(`DEV MODE next job in ${DevDelayInMS}ms at ${new Date(Date.now() + DevDelayInMS)}`)
+            await utils.timeout(DevDelayInMS)
+            this.stateMachineQueue.add({ startTime: Date.now() })
+          }
+          done()
+        }
+      }
     )
+
+    // Run the task every x time interval
+    // */5 * * * *, every 5 minutes
+    // 0 * * * *, every hour at minute 0
+    this.stateMachineQueue.add({}, { repeat: { cron: '0 * * * *' } })
```

Comment:
nice to have, switch cron to `* */n * * *` with `n` as a config `hourInterval` or smth per @dmanjunath